language-server/src/test/java/org/sonarlint/languageserver/folders/WorkspaceFoldersManagerTest.java

```diff
@@ -123,9 +124,20 @@ public void testURIAncestor() {
     assertThat(isAncestor(create("file:///foo"), create("file:///foo/bar.txt"))).isTrue();
     assertThat(isAncestor(create("file:///foo/bar"), create("file:///foo/bar.txt"))).isFalse();
     assertThat(isAncestor(create("file:///foo/bar"), create("file:///foo/bar2"))).isFalse();
+
+    // Non file scheme
+    assertThat(isAncestor(create("ftp://ftp.example.com/foo"), create("ftp://ftp.example.com/foo"))).isTrue();
+    assertThat(isAncestor(create("ftp://ftp.example.com/foo"), create("ftp://ftp.example.com/foo/bar.txt"))).isTrue();
+    assertThat(isAncestor(create("ftp://ftp.example.com/foo/bar"), create("ftp://ftp.example.com/foo/bar.txt"))).isFalse();
+    assertThat(isAncestor(create("ftp://ftp.example.com/foo/bar"), create("ftp://ftp.example.com/foo/bar2"))).isFalse();
+    assertThat(isAncestor(create("ftp://ftp.example.com/foo/bar"), create("ftp://ftp.example.com/bar.txt"))).isFalse();
+
     // Windows
-    assertThat(isAncestor(create("file://laptop/My%20Documents"), create("file://laptop/My%20Documents/FileSchemeURIs.doc"))).isTrue();
     assertThat(isAncestor(create("file:///C:/Documents%20and%20Settings/davris"), create("file:///C:/Documents%20and%20Settings/davris/FileSchemeURIs.doc"))).isTrue();
+    if (SystemUtils.IS_OS_WINDOWS) {
```

Comment:
Nitpicking: I would move the test to a standalone method and put an `assume` to skip it when the OS is not Windows.